examples/lang/erc20/src/lib.rs

```diff
@@ -152,12 +152,23 @@ contract! {
 mod tests {
     use super::*;
     use ink_core::env;
-    use std::convert::TryFrom;
+    use parity_codec::Decode;
+
+    /// Create a new AccountId with the given bytes
+    ///
+    /// Required because AccountId is a type alias and cannot be used as a constructor
+    fn new_account_id(data: [u8; 32]) -> AccountId {
+        Decode::decode(&mut data.as_ref()).expect("Valid AccountId bytes")
+    }
+
+    fn set_caller(caller: AccountId) {
```

Comment:
Maybe we can replace this by another abstraction layer around the test environment.
For example we could introduce a `ConcreteTestEnv` in the following way:

```rust
struct ConcreteTestEnv<T>
where
    T: EnvTypes,
{
    pub fn set_caller(caller: AccountId) {
        env::test::set_caller::<T>(caller)
    }
}
```
And then `ink_lang` could generate something like:
```rust
type TestEnv = ConcreteTestEnv<WhateverTheUserWantedAsTheirEnv>;
```
So they could then write
```rust
TestEnv::set_caller([0x0; 32]);
```
Instead of having to write their own wrappers around all calls.
This encapsulates the environment information into this wrapper type to that a user doesn't have to annotate all test environment calls with the same `EnvTypes` type.

Reply:
Of course: Feel free to come up with something better!

Reply:
I would prefer a solution were we could still write `env::test::set_caller` instead of `TestEnv::set_caller`.

Reply:
Let me know what you think of the solution in https://github.com/paritytech/ink/pull/108/commits/5c2158c33cf54a4e2ec3f8119f6d0d15340bb240